src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1146,16 +1146,3 @@ def validator_delete_retention_days(namespace):
     if namespace.enable_delete_retention is False and namespace.delete_retention_days is not None:
         raise ValueError(
             "incorrect usage: '--delete-retention-days' is invalid when '--enable-delete-retention' is set to false")
-
-    if namespace.enable_delete_retention is None and namespace.delete_retention_days is not None:
-        raise ValueError(
-            "incorrect usage: please specify '--enable-delete-retention true' if you want to set the value for "
-            "'--delete-retention-days'")
-
-    if namespace.delete_retention_days or namespace.delete_retention_days == 0:
-        if namespace.delete_retention_days < 1:
-            raise ValueError(
-                "incorrect usage: '--delete-retention-days' must be greater than or equal to 1")
-        if namespace.delete_retention_days > 365:
-            raise ValueError(
-                "incorrect usage: '--delete-retention-days' must be less than or equal to 365")
```

Comment:
The reason why this verification was added here is that the verification of SDK is not correct, and the range of verification prompt is [2, 364].            
If you want to remove this verification logic, please make sure that they have fixed the error range prompt.

The previous errors of SDK were as follows:
> validation error: Parameter 'DeleteRetentionPolicy.days' must be less than 365.  
   validation error: Parameter 'DeleteRetentionPolicy.days' must be greater than 1.

Reply:
I think it will be better to make server side give days range check. Otherwise, if server changes the days range, we will block users and currently service error message is good enough to me.
But if you have any concern, feel free to let me know.

Reply:
Because the error message of the SDK validation is not correct, so we may need to verify it in advance to provide the correct error message.
It is a problem about msrest in essence, and I suspect they are unlikely to fix it in a short time:
https://github.com/Azure/msrest-for-python/issues/191
If they can fix this problem, I think it's ok to remove this check.